src/mbgl/style/style_parser.cpp

```diff
@@ -36,20 +36,30 @@ void StyleParser::parse(JSVal document) {
     if (document.HasMember("layers")) {
         parseLayers(document["layers"]);
 
+        // create shape annotations source
```

Comment:
Instead of adding these fake layers to the Style when parsing, we should add a facility to add these layers to the Map object itself so that we can still switch styles without losing all of our annotations.